StrawberryShake.Blazor causes thousands of build errors

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

Just creating a simple Blazor WASM project in .NET 7 and adding StrawberryShake.Blazor causes a bunch of errors when I try to build it. The first thing that happens is that I get a warning, and if I ctrl+c the console, I'm able to copy that warning. It seems to indicate that the build is enumerating my ENTIRE DRIVE, which is crazy:

```
MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "C:\Users\evshe\.nuget\packages\strawberryshake.blazor\13.1.0\build\Straw
berryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always def 
ined. [C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj]
```

Also, when I added it originally in VS2022, it started downloading tons of files from OneDrive into a Unity folder which was in a completely separate place than this project. It froze up VS and I had to completely redo a lot of things. My guess is that this drive enumeration is causing it to try to download ALL my OneDrive files, which is not acceptable. The only way I was able to safely reproduce the issue was in VS Code with CLI commands.

### Expected: **Do not enumerate all files in my drive**

### Steps to reproduce

1. Create a new Blazor WASM project and add StrawberryShake.Blazor, then build it:
```
dotnet new blazorwasm -o TestBlazorGraphQL
cd TestBlazorGraphQL
dotnet add package StrawberryShake.Blazor
dotnet build
```


### Relevant log output

```shell
Warning (if Ctrl+c):

MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "C:\Users\evshe\.nuget\packages\strawberryshake.blazor\13.1.0\build\StrawberryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined. [C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj]

End of error output if I let it go:

C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\ItemTemplates\Web\CSharp\1033\DynamicDataWebSiteField.cs\DynamicData\FieldTemplates\CodeBeside_Edit.cs(37,1): error CS1022: Type or namespace definition, or end-of-file expected [C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj]
    5 Warning(s)
    15077 Error(s)
```


### Additional Context?

_No response_

### Version

13.1.0

Just to rule out any possible environmental issues with my machine, I used docker and got the same result (I also added the tool manifest and the Strawberry Shake tools, but it really doesn't make any difference). Run from the app directory:
```
docker run --rm -v ${PWD}:/app mcr.microsoft.com/dotnet/sdk:7.0 dotnet build /app
```

Same issue, although the errors were a lot smaller, since it is a container:
```
user@machine TestBlazorGraphQL % docker run --rm -v ${PWD}:/app mcr.microsoft.com/dotnet/sdk:7.0 dotnet build /app   
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  Restored /app/TestBlazorGraphQL.csproj (in 5.09 sec).
  Generate C# Clients started.
  Generate C# Clients completed in 57 ms
  TestBlazorGraphQL -> /app/bin/Debug/net7.0/TestBlazorGraphQL.dll
  TestBlazorGraphQL (Blazor output) -> /app/bin/Debug/net7.0/wwwroot

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:15.49
user@machine TestBlazorGraphQL % docker run --rm -v ${PWD}:/app mcr.microsoft.com/dotnet/sdk:7.0 dotnet build /app
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  Restored /app/TestBlazorGraphQL.csproj (in 4.64 sec).
/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets(38,20): error MSB4184: The expression "[MSBuild]::VersionGreaterThanOrEquals('', 6)" cannot be evaluated. Version string was not in a correct format. [/app/TestBlazorGraphQL.csproj]
MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined. [/app/TestBlazorGraphQL.csproj]
CSC : error CS2021: File name '/**/*.cs' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long [/app/TestBlazorGraphQL.csproj]
CSC : error CS2001: Source file '/**/*.cs' could not be found. [/app/TestBlazorGraphQL.csproj]

Build FAILED.

MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined. [/app/TestBlazorGraphQL.csproj]
/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets(38,20): error MSB4184: The expression "[MSBuild]::VersionGreaterThanOrEquals('', 6)" cannot be evaluated. Version string was not in a correct format. [/app/TestBlazorGraphQL.csproj]
CSC : error CS2021: File name '/**/*.cs' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long [/app/TestBlazorGraphQL.csproj]
CSC : error CS2001: Source file '/**/*.cs' could not be found. [/app/TestBlazorGraphQL.csproj]
    1 Warning(s)
    3 Error(s)

Time Elapsed 00:00:07.23
```

--- a/src/StrawberryShake/MetaPackages/Common/MSBuild/StrawberryShake.targets
+++ b/src/StrawberryShake/MetaPackages/Common/MSBuild/StrawberryShake.targets
@@ -18,10 +18,17 @@
   <Target
     Name="GenerateGraphQLCode"
     BeforeTargets="BeforeBuild"
-    DependsOnTargets="_GenerateGraphQLCode">
+    DependsOnTargets="_GraphQLCodeGenerationRoot; _GenerateGraphQLCode">
     <ItemGroup>
       <Compile Include="$(GraphQLCodeGenerationRoot)\**\*.cs" />
     </ItemGroup>
+  </Target>
+
+  <Target
+    Name="_GraphQLCodeGenerationRoot">
+    <PropertyGroup>
+      <GraphQLCodeGenerationRoot>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)berry'))</GraphQLCodeGenerationRoot>
+    </PropertyGroup>
   </Target>
 
   <Target Name="_ParseVersion">
@@ -34,7 +41,7 @@
     Name="_GenerateGraphQLCode"
     Inputs="@(GenerateGraphQLCodeItems)"
     Outputs="$(MSBuildProjectDirectory)\$(IntermediateOutputPath)berry\.build.info"
-    DependsOnTargets="_ParseVersion"
+    DependsOnTargets="_ParseVersion;_GraphQLCodeGenerationRoot"
     Condition="'@(GraphQL)' != ''">
 
     <PropertyGroup>
@@ -47,7 +54,6 @@
       Condition="'$(DotNetMajor)' != '6' AND '$(DotNetMajor)' != '7'" />
 
     <PropertyGroup>
-      <GraphQLCodeGenerationRoot>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)berry'))</GraphQLCodeGenerationRoot>
       <GraphQLQueryGenerationRoot>$(MSBuildProjectDirectory)\$(GraphQLPersistedQueryOutput.TrimEnd('/').TrimEnd('\'))</GraphQLQueryGenerationRoot>
       <GenTool Condition="'$(DotNetMajor)' == '6'">$([System.IO.Path]::Combine($(MSBuildThisFileDirectory), "..", "tools", "net6", "dotnet-graphql.dll"))</GenTool>
       <GenTool Condition="'$(DotNetMajor)' == '7'">$([System.IO.Path]::Combine($(MSBuildThisFileDirectory), "..", "tools", "net7", "dotnet-graphql.dll"))</GenTool>

